File: sdks/java/src/main/java/com/rodo/RodoTest.java
package com.rodo;

import org.junit.Test;
import java.io.File;
import static org.junit.Assert.*;

public class RodoTest {
    @Test
    public void testSetGet() throws Exception {
        String path = "test_java.rd";
        Rodo r = Rodo.link(path);
        Rodo.Value v = new Rodo.Value();
        v.type = Rodo.TYPE_STRING;
        v.stringVal = "Davi";
        r.set("nome", v);
        Rodo.Value got = r.get("nome");
        assertNotNull(got);
        assertEquals("Davi", got.stringVal);
        assertTrue(r.has("nome"));
        r.close();
        new File(path).delete();
    }

    @Test
    public void testPersistence() throws Exception {
        String path = "test_persist_java.rd";
        {
            Rodo r = Rodo.link(path);
            Rodo.Value v = new Rodo.Value();
            v.type = Rodo.TYPE_INT;
            v.intVal = 87;
            r.set("level", v);
            r.close();
        }
        {
            Rodo r = Rodo.link(path);
            Rodo.Value got = r.get("level");
            assertNotNull(got);
            assertEquals(87, got.intVal);
            r.close();
        }
        new File(path).delete();
    }

    @Test
    public void testDelete() throws Exception {
        String path = "test_delete_java.rd";
        Rodo r = Rodo.link(path);
        Rodo.Value v = new Rodo.Value();
        v.type = Rodo.TYPE_INT;
        v.intVal = 1500;
        r.set("coins", v);
        r.delete("coins");
        assertFalse(r.has("coins"));
        r.close();
        new File(path).delete();
    }
}

File: sdks/go/rodo_test.go
package rodo

import (
	"testing"
	"os"
)

func TestSetGet(t *testing.T) {
	path := "test_go.rd"
	r, err := Link(path)
	if err != nil {
		t.Fatal(err)
	}
	defer os.Remove(path)

	val := Value{Type: TypeString, StringVal: "Davi"}
	if err := r.Set("nome", val); err != nil {
		t.Fatal(err)
	}
	got, ok := r.Get("nome")
	if !ok || got.StringVal != "Davi" {
		t.Fatal("get failed")
	}
	if !r.Has("nome") {
		t.Fatal("has failed")
	}
	if err := r.Close(); err != nil {
		t.Fatal(err)
	}
}

func TestPersistence(t *testing.T) {
	path := "test_persist_go.rd"
	{
		r, err := Link(path)
		if err != nil {
			t.Fatal(err)
		}
		r.Set("level", Value{Type: TypeInt, IntVal: 87})
		r.Close()
	}
	{
		r, err := Link(path)
		if err != nil {
			t.Fatal(err)
		}
		got, ok := r.Get("level")
		if !ok || got.IntVal != 87 {
			t.Fatal("data did not persist")
		}
		r.Close()
	}
	os.Remove(path)
}

func TestDelete(t *testing.T) {
	path := "test_delete_go.rd"
	r, _ := Link(path)
	r.Set("coins", Value{Type: TypeInt, IntVal: 1500})
	if err := r.Delete("coins"); err != nil {
		t.Fatal(err)
	}
	if r.Has("coins") {
		t.Fatal("key not deleted")
	}
	r.Close()
	os.Remove(path)
}